yarn-project/foundation/src/abi/decoder.ts

```diff
@@ -86,3 +86,44 @@ class ReturnValuesDecoder {
 export function decodeReturnValues(abi: FunctionAbi, returnValues: Fr[]) {
   return new ReturnValuesDecoder(abi, returnValues.slice()).decode();
 }
+
+/**
+ * Decodes the signature of a function from the name and parameters.
+ */
+export class FunctionSignatureDecoder {
+  constructor(private name: string, private parameters: ABIParameter[]) {}
```

Comment:
I know this is a matter of personal preference, but I prefer to export plain functions rather than classes with a "do" method. A user of this API just wants to `decodeFunctionSignature(name, parameters)`, and doesn't care about the `FunctionSignatureDecoder` that's doing the job.

Reply:
True, mainly did it like this as since it just followed the other decoder.